src/main/java/net/wasdev/wlp/common/plugins/util/InstallFeatureUtil.java

```diff
@@ -0,0 +1,560 @@
+
+        try {
+            Map<String, Object> mapBasedInstallKernel = createMapBasedInstallKernelInstance(installDirectory);
+            mapBasedInstallKernel.put("install.local.esa", true);
+            mapBasedInstallKernel.put("single.json.file", jsonRepos);
+            mapBasedInstallKernel.put("features.to.resolve", featuresToInstall);
+            mapBasedInstallKernel.put("license.accept", isAcceptLicense);
+
+            if (isDebugEnabled()) {
+                mapBasedInstallKernel.put("debug", Level.FINEST);
+            }
+
+            Collection<?> resolvedFeatures = (Collection<?>) mapBasedInstallKernel.get("action.result");
+            if (resolvedFeatures == null) {
+                debug("action.exception.stacktrace: "+mapBasedInstallKernel.get("action.exception.stacktrace"));
+                String exceptionMessage = (String) mapBasedInstallKernel.get("action.error.message");
+                throw new PluginExecutionException(exceptionMessage);
+            } else if (resolvedFeatures.isEmpty()) {
+                debug("action.exception.stacktrace: "+mapBasedInstallKernel.get("action.exception.stacktrace"));
+                String exceptionMessage = (String) mapBasedInstallKernel.get("action.error.message");
+                if (exceptionMessage.contains("CWWKF1250I")){
+                    warn(exceptionMessage);
+                    return;
+                } else {
+                    throw new PluginExecutionException(exceptionMessage);
+                }
+            }
+            Collection<File> artifacts = downloadEsas(resolvedFeatures);
+
+            for (File esaFile: artifacts ){
+                mapBasedInstallKernel.put("license.accept", isAcceptLicense);
+                mapBasedInstallKernel.put("action.install", esaFile);
+                Integer ac = (Integer) mapBasedInstallKernel.get("action.result");
+                debug("action.result: "+ac);
+                debug("action.error.message: "+mapBasedInstallKernel.get("action.error.message"));
+                if (mapBasedInstallKernel.get("action.error.message") != null) {
+                    debug("action.exception.stacktrace: "+mapBasedInstallKernel.get("action.exception.stacktrace"));
+                    String exceptionMessage = (String) mapBasedInstallKernel.get("action.error.message");
+                    debug(exceptionMessage);
+                }
+            }
+        } catch (PrivilegedActionException e) {
+            throw new PluginExecutionException("Could not load the jar " + getMapBasedInstallKernelJar(installDirectory).getAbsolutePath(), e);
+        }
+    }
+    
+    private Map<String, Object> createMapBasedInstallKernelInstance(File installDirectory) throws PrivilegedActionException {
+        final File installJarFile = getMapBasedInstallKernelJar(installDirectory);
+        String installJarFileSubpath = installJarFile.getParentFile().getName() + File.separator + installJarFile.getName();
```

Comment:
Add a null check on `installJarFile` since `getMapBasedInstallKernelJar` can return null.

Reply:
Actually I realize you do the null check in the constructor. In that case can you set `installJarFile` as a class variable so that you don't have to call it throughout?